agent/statechange/statechange.go

```diff
@@ -25,6 +25,10 @@ const (
 	// AttachmentEvent is used to define the attachment state transition events
 	// emitted by ENI watcher
 	AttachmentEvent
+
+	// ManagedAgentEvent is used to define the managed agent state transition events
+	// emitted by the engine
+	ManagedAgentEvent
```

Comment:
glad to see this differentiation between `ContainerEvent` and `ManagedAgentEvent`